server/core/basic_cluster.go

```diff
@@ -151,25 +152,47 @@ func (bc *BasicCluster) UpdateStoreStatus(storeID uint64, leaderCount int, regio
 	bc.Stores.UpdateStoreStatus(storeID, leaderCount, regionCount, pendingPeerCount, leaderSize, regionSize)
 }
 
+const randomRegionMaxRetry = 10
+
 // RandFollowerRegion returns a random region that has a follower on the store.
 func (bc *BasicCluster) RandFollowerRegion(storeID uint64, opts ...RegionOption) *RegionInfo {
 	bc.RLock()
-	defer bc.RUnlock()
-	return bc.Regions.RandFollowerRegion(storeID, opts...)
+	regions := bc.Regions.RandFollowerRegions(storeID, randomRegionMaxRetry)
```

Comment:
BTW, we need to ensure there is one or more region passed check option

Reply:
Can't ensure that. It's possible that no region can pass checkers. Old implementation only try 10 times too.